ReSwift/CoreTypes/Store.swift

```diff
@@ -40,11 +40,29 @@ open class Store<State: StateType>: StoreType {
 
     private var isDispatching = false
 
+    /// Indicates if new subscriptions attempt to apply `skipRepeats` 
+    /// by default.
+    let subscriptionsAutomaticallySkipRepeats: Bool
+
+    /// Initializes the store with a reducer, an initial state and a list of middleware.
+    ///
+    /// Middleware is applied in the order in which it is passed into this constructor.
+    ///
+    /// - parameter reducer: Main reducer that processes incomind actions.
+    /// - parameter state: Initial state, if any. Can be `nil` and will be 
+    ///   provided by the reducer in that case.
+    /// - parameter middleware: Ordered list of action pre-processors, acting 
+    ///   before the root reducer.
+    /// - parameter automaticallySkipsRepeats: If `true`, the store will attempt 
+    ///   to skip idempotent state updates when a subscriber's state type 
+    ///   implements `Equatable`. Defaults to `false`.
     public required init(
         reducer: @escaping Reducer<State>,
         state: State?,
-        middleware: [Middleware<State>] = []
+        middleware: [Middleware<State>] = [],
+        automaticallySkipsRepeats: Bool = false
```

Comment:
The changes look fine to me, my only question would be whether we want to change the default on users.
It might be better to have the default be `true`, which maintains backwards compatibility with no change required by users, and yet allows opt-out for those who wish.